pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -61,6 +62,266 @@ func newGitHubScraper(
+
+		for i := 0; i < bp; i++ {
+			r, err := getBranchData(ctx, client, repoName, ghs.cfg.GitHubOrg, 50, defaultBranch, branchCursor)
+			if err != nil {
+				ghs.logger.Sugar().Errorf("error getting branch data", zap.Error(err))
+			}
+
+			branches = append(branches, r.Repository.Refs.Nodes...)
+
+			branchCursor = &r.Repository.Refs.PageInfo.EndCursor
+		}
+		branchCh <- branches
+	}
+}
+
+func (ghs *githubScraper) processBranches(
+	ctx context.Context,
+	client graphql.Client,
+	now pcommon.Timestamp,
+	branchCh chan []BranchNode,
+) {
+
+	for branches := range branchCh {
+		for _, branch := range branches {
+			if branch.Name == branch.Repository.DefaultBranchRef.Name || branch.Compare.BehindBy == 0 {
+				continue
+			}
+			repoName := branch.Repository.Name
+			ghs.logger.Sugar().Debugf(
+				"default branch behind by: %d\n %s branch behind by: %d in repo: %s",
+				branch.Compare.BehindBy, branch.Name, branch.Compare.AheadBy, repoName)
+
+			// Yes, this looks weird. The aheadby metric is referring to the number of commits the branch is AHEAD OF the
+			// default branch, which in the context of the query is the behind by value. See the above below comment about
+			// BehindBy vs AheadBy.
+			ghs.mb.RecordGitRepositoryBranchCommitAheadbyCountDataPoint(now, int64(branch.Compare.BehindBy), repoName, branch.Name)
+			ghs.mb.RecordGitRepositoryBranchCommitBehindbyCountDataPoint(now, int64(branch.Compare.AheadBy), repoName, branch.Name)
+
+			// We're using BehindBy here because we're comparing against the target
+			// branch, which is the default branch. In essence the response is saying
+			// the default branch is behind the queried branch by X commits which is
+			// the number of commits made to the queried branch but not merged into
+			// the default branch. Doing it this way involves less queries because
+			// we don't have to know the queried branch name ahead of time.
+			cp := getNumPages(float64(100), float64(branch.Compare.BehindBy))
+			ghs.processCommits(ctx, client, repoName, now, cp, branch)
+		}
+	}
+	//close(branchCh)
+}
+
+func (ghs *githubScraper) getContributorCount(
+	ctx context.Context,
+	client graphql.Client,
+	repos []SearchNode,
+	now pcommon.Timestamp,
+	waitGroup *sync.WaitGroup,
+) {
+	defer waitGroup.Done()
+
+	gc := github.NewClient(ghs.client)
+	for _, repo := range repos {
+		var repoName string
+
+		if n, ok := repo.(*SearchNodeRepository); ok {
+			repoName = n.Name
+		}
+
+		contribs, _, err := gc.Repositories.ListContributors(ctx, ghs.cfg.GitHubOrg, repoName, nil)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting contributor count", zap.Error(err))
+		}
+
+		contribCount := 0
+		if len(contribs) > 0 {
+			contribCount = len(contribs)
+		}
+
+		ghs.logger.Sugar().Debugf("contributor count: %v for repo %v", contribCount, repoName)
+		ghs.mb.RecordGitRepositoryContributorCountDataPoint(now, int64(contribCount), repoName)
+	}
+}
+
+func chunkSlice[T any](slice []T, chunkSize int) [][]T {
```

Comment:
Let's pop this over into helper & write a test for it. Seems like an easily testable function.